[psh12320] iP

psh12320 opened the pull request:
# This is Taskalyn!

> "The key is not to prioritize what's on your schedule, but to schedule your priorities." - Stephen Covey ([source](https://www.forbes.com/sites/kevinkruse/2012/07/16/the-7-habits/#:~:text=The%20key%20is%20not%20to%20prioritize%20what%27s%20on%20your%20schedule%2C%20but%20to%20schedule%20your%20priorities.))

**Taskalyn is here to help you manage your tasks!** 😊

- Add Todo, Deadline, Event Tasks.
- Mark as complete or incomplete.
- Find tasks with the same keyword.
- ~~FAST~~ ***SUPER FAST*** to use.

All you have to do is:
1. Download the file [here](https://nus-cs2103-ay2425s1.github.io/website/admin/standardsAndConventions.html).
2. Add your tasks.
3. Let it manage your tasks for you!

**Here are Taskalyn's features:**
- [x] Managing Todo
- [x] Managing Deadlines
- [x] Managing Events 

If you are a programmer, you can use it to practice Java too. Here's the `main` method. 
``` java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

psh12320 committed:
update .gitignore to ignore .class files

psh12320 committed:
updated .gitignore to ignore .class files in subdirectories too

psh12320 committed:
Rename, Greet, Exit

psh12320 committed:
importing scanner

psh12320 committed:
functionality for Level-2 finished

psh12320 committed:
removed extra spaces in between lines

psh12320 committed:
finished part of Level 2

psh12320 committed:
added TaskManager class

psh12320 committed:
added the word added when adding task

psh12320 committed:
added a new Task class

psh12320 committed:
complete and incomplete methods added in TaskManager

psh12320 committed:
added an error flag to revert to default case

psh12320 committed:
fix cases of input starting with mark and more than 3 words

psh12320 committed:
created a TodoTask class

psh12320 committed:
created a DeadlineTask class

psh12320 committed:
created an EventTask class

psh12320 committed:
changed Task class into an abstract class

psh12320 committed:
added functionality for ToDo, Deadline, Event

psh12320 committed:
used i/o redirection for testing

psh12320 committed:
changed EXPECTED.TXT accordingly

psh12320 committed:
added new exception classes

psh12320 committed:
exceptions added to Taskalyn.java

psh12320 committed:
delete feature added

psh12320 committed:
changing output when list has zero tasks

psh12320 committed:
Add Database.java to store tasks

psh12320 committed:
Update TaskManager.java with Database object

psh12320 committed:
Merge branch-Level-7 into master

psh12320 committed:
Update DeadlineTask.java and TaskManager.java with LocalDateTime objects

psh12320 committed:
Merge branch-Level-8 into master

psh12320 committed:
Add Ui.java and Parser.java

psh12320 committed:
Add package taskalyn

psh12320 committed:
Merge branch 'add-gradle-support'

psh12320 committed:
Add UiTest.java and ParserTest.java

psh12320 committed:
Update ParserTest.java

psh12320 committed:
Create JAR file

BunnyHoppp commented on src/main/java/taskalyn/Database.java:
```diff
@@ -0,0 +1,58 @@
+package taskalyn;
+
+import java.nio.file.Files;
```
I like that you imported all the statements explicitly

BunnyHoppp commented on src/main/java/taskalyn/Database.java:
```diff
@@ -0,0 +1,58 @@
+import java.util.List;
+
+public class Database {
+    private static final String DIRECTORY = "./data";
```
I like the use of SCREAMING_SNAKE_CASE for constants

psh12320 replied:
Thanks!

BunnyHoppp commented on src/main/java/taskalyn/TaskManager.java:
```diff
@@ -0,0 +1,133 @@
+        updateDatabase();
+    }
+
+    public void incompleteTask(int taskId) {
```
Should the name of the method be a verb here instead?

psh12320 replied:
You're right. Will do.

BunnyHoppp commented on src/main/java/taskalyn/TaskManager.java:
```diff
@@ -0,0 +1,133 @@
+    }
+
+    private Task findTask(String line) {
+        String[] parts = line.split(" \\| ");
```
I like that your list variables are plural

psh12320 replied:
Thanks!

BunnyHoppp commented on src/main/java/taskalyn/Taskalyn.java:
```diff
@@ -0,0 +1,21 @@
+package taskalyn;
+
```
Perhaps we can add some header comments for some classes?

psh12320 replied:
Yes that will be helpful. I will do so.

BunnyHoppp commented on src/main/java/taskalyn/Parser.java:
```diff
@@ -0,0 +1,153 @@
+        this.ui = ui;
+        this.taskManager = taskManager;
+    }
+    public boolean parse(TaskManager taskManager) {
```
I like how clean this code segment is in this method!

psh12320 replied:
Thanks!

BunnyHoppp reviewed (commented):
I like the overall style

psh12320 committed:
Update ParserTest.java

psh12320 committed:
Update build.gradle

psh12320 committed:
Add new test in ParserTest.java

psh12320 committed:
Update readCommand method in Ui.java

psh12320 committed:
Update UiTest.java

psh12320 committed:
Update ParserTest.java

justsparsh commented on src/main/java/taskalyn/Database.java:
```diff
@@ -0,0 +1,58 @@
+
+    public Database() {
+        try {
+            List<String> txtLines = Files.readAllLines(FILE_PATH);
```
Perhaps you can rename this variable as `textLines` to avoid unnecessary abbreviations

psh12320 replied:
You're right! Thanks.

justsparsh commented:
Perhaps you can enhance your packaging by grouping together files of similar types (e.g. Tasks, Commands etc.)

justsparsh commented on src/main/java/taskalyn/TaskManager.java:
```diff
@@ -0,0 +1,129 @@
+        return null;
+    }
+
+    private void updateDatabase() {
```
Perhaps you could add JavaDocs for this private method too as it is crucial in this class

psh12320 replied:
That's true. I'll add them.

psh12320 committed:
Update Database.java

psh12320 committed:
Add TaskManagerTest.java

psh12320 committed:
Add MockDatabase.java

psh12320 committed:
Update getDatabaseSize method in Database.java

psh12320 committed:
Add DatabaseTest.java

psh12320 committed:
Update JavaDoc comments across all files

psh12320 committed:
Merge branch 'branch-A-JavaDoc'

psh12320 committed:
Fix coding standard violations across multiple files

psh12320 committed:
Merge branch-A-CodingStandard into master

psh12320 committed:
Fix coding standard violations across multiple files

psh12320 committed:
Add searchTasksByKeyword method in TaskManager.java

psh12320 committed:
Add find command in Parser.java

psh12320 committed:
Rename getTaskInfo to getTaskDescription in Task.java

psh12320 committed:
Merge branch-Level-9 into master

psh12320 committed:
Fix unit test failures across test files

psh12320 committed:
Add a GUI to Taskalyn

psh12320 committed:
Fix GUI issues

psh12320 committed:
Update JUnit tests in ParserTest.java and UiTest.java

psh12320 committed:
Update to getTaskDescription method in Task subclasses

psh12320 committed:
Update GUI with pictures and style

psh12320 committed:
Merge branch-Level-10 to master

psh12320 committed:
Add checkstyle.xml and suppressions.xml

psh12320 committed:
Merge branch 'branch-A-CheckStyle'

psh12320 committed:
Fix coding style across multiple files

psh12320 committed:
Fix JUnit test cases and coding style violations

psh12320 committed:
Add assertions in Parser.java and TaskManager.java

psh12320 committed:
Merge pull request #1 from psh12320/branch-A-Assertions

Add assertions in Parser.java and TaskManager.java

psh12320 committed:
Merge branch 'master' into branch-A-CodeQuality

psh12320 committed:
Merge branch 'master' into branch-A-Streams

psh12320 committed:
Add welcome message and delayed exit in MainWindow.java

The application currently does not show a welcome message and
the bye command does not close the application.

Let's
* update initializate method in MainWindow to show welcome message
* include a 5 second timer before exiting application after user
types in the bye command.

This will provide a smoother interaction experience for the user.

psh12320 committed:
Add enums class and abstract command handling in Parser.java

The Parser class previously relied on magic literals to
represent command types as Strings.

This commit introduces an enums class, CommandType, to
prevent magic literals.

The handleTaskOperation method in Parser.java was abstracted
to remove duplicated logic in delete, mark, and unmark
commands.

psh12320 committed:
Update JUnit tests across multiple files

psh12320 committed:
Improve code quality across multiple files

psh12320 committed:
Merge pull request #2 from psh12320/branch-A-CodeQuality

Branch a code quality

psh12320 committed:
Merge branch 'master' into branch-A-Streams

psh12320 committed:
Update Parser.java with Java Streams

psh12320 committed:
Merge pull request #3 from psh12320/branch-A-Streams

Branch a streams

psh12320 committed:
Add sort command to sort deadline tasks chronologically

psh12320 committed:
Update UI messages and JUnit test cases

psh12320 committed:
Merge pull request #4 from psh12320/branch-C-Sort

Branch c sort

psh12320 committed:
Add gradle.yml for CI

psh12320 committed:
Update user.png with new photo dimensions

psh12320 committed:
Update README.md with a user guide

psh12320 committed:
Fix typo in README.md

psh12320 committed:
Fix code block typo in README.md

psh12320 committed:
Add zero-width space char in README.md

psh12320 committed:
Update README.md with non breaking space char

psh12320 committed:
Fix code block indent issue in README.md

psh12320 committed:
Fix image alignment in README.md

psh12320 committed:
Update README.md with new image size

psh12320 committed:
Update table in README.md

psh12320 committed:
Update README.md with footnote

psh12320 committed:
Fix bug in table in README.md

psh12320 committed:
Remove IO redirection tests from CI workflow

psh12320 committed:
Update Javadoc across multiple files

psh12320 committed:
Add JUnit tests in ParserTest and TaskManagerTest

psh12320 committed:
Add more error handlers in Parser.java

psh12320 committed:
Update javadoc and comments across multiple files

psh12320 committed:
Fix bugs in DatabaseTest and TaskManagerTest

psh12320 committed:
Improve code quality across multiple files

psh12320 committed:
Add JUnit test cases in ParserTest.java

psh12320 committed:
Improve code quality across multiple files

psh12320 committed:
Update user image and README.md

psh12320 committed:
Update README.md in the root directory
